disperser/batcher/metrics.go

```diff
@@ -249,3 +250,42 @@ func (t *TxnManagerMetrics) UpdateTxQueue(txQueue int) {
 func (t *TxnManagerMetrics) IncrementTxnCount(state string) {
 	t.NumTx.WithLabelValues(state).Inc()
 }
+
+type DynamoDBCollector struct {
+	blobMetadataStore *blobstore.BlobMetadataStore
+	blobStatusMetric  *prometheus.Desc
+	logger            common.Logger
+}
+
+func NewDynamoDBCollector(blobMetadataStore *blobstore.BlobMetadataStore, logger common.Logger) *DynamoDBCollector {
+	return &DynamoDBCollector{
+		blobMetadataStore: blobMetadataStore,
+		blobStatusMetric: prometheus.NewDesc("dynamodb_blob_metadata_status_count",
+			"Number of blobs with specific status in DynamoDB",
+			[]string{"status"},
+			nil,
+		),
+		logger: logger,
+	}
+}
+
+func (collector *DynamoDBCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- collector.blobStatusMetric
+}
+
+func (collector *DynamoDBCollector) Collect(ch chan<- prometheus.Metric) {
+	metadatas, err := collector.blobMetadataStore.GetBlobMetadataByStatus(context.Background(), disperser.Processing)
```

Comment:
How long would this take and how often it'll be called?

What about other status?

Reply:
Default scrape interval is 60s. This specific call is really fast because the number of processing metadata should never exceed 1000 in the happy case (unless the batcher is down).

For the other status, I would say # of metadata processing is the most important one because if it crosses a threshold, we know that blobs are being backed up.

I took a sample of values just now on the testnet:
Processing - milliseconds
{
  "Count": 36,
  "ScannedCount": 36,
  "ConsumedCapacity": null
}

Confirmed - milliseconds
{
  "Count": 1288,
  "ScannedCount": 1288,
  "ConsumedCapacity": null
}

Failed - milliseconds
{
  "Count": 5712,
  "ScannedCount": 5712,
  "ConsumedCapacity": null
}

Finalized - Takes too long, doesn't finish.

InsufficientSignatures - Takes a couple of seconds
{
  "Count": 58685,
  "ScannedCount": 58685,
  "ConsumedCapacity": null
}

Reply:
Hmm just realized that if the batcher is down then the metric won't get emitted. Maybe this should go in the data server?

Reply:
Yes the processing state is more useful.
Sounds good to put this on dataapi server since it's like a "side observability".